### PR TITLE
chore(aws_sqs sink): Migrate to AWS SDK

### PR DESCRIPTION
**Pull request by fuchsnj**
This migrates the aws_sqs sink to use the new AWS SDK.

### CODE REVIEW

**Review by jszwedko — APPROVED** [review]
NIce!

File: src/sinks/aws_sqs.rs [hunk 1]
```diff
@@ -148,25 +185,23 @@ impl SinkConfig for SqsSinkConfig {
...
+            .send()
             .await
             .map(|_| ())
             .context(GetQueueAttributesSnafu)
             .map_err(Into::into)
     }
 
-    pub fn create_client(&self, proxy: &ProxyConfig) -> crate::Result<SqsClient> {
-        let region = (&self.region).try_into()?;
-        let tls_settings = MaybeTlsSettings::from(TlsSettings::from_options(&self.tls)?);
```
Comment (pablosichert) [inline src/sinks/aws_sqs.rs:163]: I don't see `self.tls` referenced any more, is that intentional? Can we drop the `tls` field from `SqsSinkConfig`?

Comment (jszwedko) [reply src/sinks/aws_sqs.rs:163]: Oof, good catch. We should preserve the `tls` settings. Users do use that when using AWS API-compatible tools. I think this needs to be fixed for the other AWS components that were migrated, as well. We can probably just add it as another argument to `create_client`?

Comment (fuchsnj) [reply src/sinks/aws_sqs.rs:163]: I'll look at this more. I might need to add support for these in the client builder

**Review by jszwedko — APPROVED** [review]
Looks good, thanks for adding the TLS handling! And thanks to @pablosichert for catching it.

**Review by pablosichert — APPROVED** [review]
Looks pretty straight-forward 👍

**Pull request merged**